src/unit_frame.cpp

```diff
@@ -649,10 +649,8 @@ void unit_frame::redraw(const int frame_time,bool on_start_time,bool in_scope_of
 			sound::play_sound(current_data.sound);
 		}
 		if(!current_data.text.empty()  ) {
-			game_display::get_singleton()->float_label(src,current_data.text,
-			(current_data.text_color & 0x00FF0000) >> 16,
-			(current_data.text_color & 0x0000FF00) >> 8,
-			(current_data.text_color & 0x000000FF) >> 0);
+			game_display::get_singleton()->float_label(src, current_data.text,
+				int_to_color(current_data.text_color));
```

Comment:
Ah, looks like it [should be fine](https://github.com/Vultraz/wesnoth/blob/804d1eae83ba13baafbe429452d81f060a008714/src/sdl/utils.cpp#L63).